### PR TITLE
Updates Bundle Observation Vector to support Csm observations.

### PR DESCRIPTION
**Pull request by krlberry**
## Description
* Updates Bundle Observation Vector to support Csm observations.

This is ready for review / eventual merge. 

## Related Issue
#4410 

## Motivation and Context
See #4410 

## How Has This Been Tested?
ctest -R Jigsaw && ctest -R jigsaw && ctest -R Bundle

Tests pass!

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ ] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp [hunk 1]
```diff
@@ -117,20 +120,36 @@ namespace Isis {
...
-                                                                 instrumentId,
-                                                                 bundleSettings->bundleTargetBody());
 
+      bool isis = true;
+      if (bundleImage->camera() != NULL) {
+        isis = bundleImage->camera()->GetCameraType() != Camera::Csm;
+      }
+
+      BundleObservation *isisObservation = NULL;
+      CsmBundleObservation *csmObservation = NULL;
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp]: There's no need to do this. You can just use an Abstract observation.

File: isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp [hunk 2]
```diff
@@ -148,16 +167,28 @@ namespace Isis {
...
 
-      isisObservation->setSolveSettings(solveSettings);
-
-      bundleObservation.reset(isisObservation);
+      if (isis) {
+        isisObservation->setSolveSettings(solveSettings);
+        bundleObservation.reset(isisObservation);
+      }
+      else {
+//        csmObservation->setSolveSettings(solveSettings); doesn't exist
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp]: This does exist, see CsmBundleObservation::setSolveSettings

**Review by jessemapel — COMMENTED** [review]
Because the ISIS and CSM observation API are so much closer together now, I am not a big fan of the two vectors solution. Can we instead initialize the exterior orientation and body rotation in the addNew function? They get called immediately after it anyway.

There's even a TODO in the current code for this:

https://github.com/USGS-Astrogeology/ISIS3/blob/dev/isis/src/control/objs/BundleAdjust/BundleAdjust.cpp#L438

File: isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp (hunk 2 above)
Comment (krlberry) [reply isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp]: Thanks @jessemapel forgot to update that. It didn't exist, I think, before the merge.

File: isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp [hunk 3]
```diff
@@ -117,20 +118,37 @@ namespace Isis {
...
     else {
       // create new BundleObservation and append to this vector
-      BundleObservation *isisObservation = new BundleObservation(bundleImage,
-                                                                 observationNumber,
-                                                                 instrumentId,
-                                                                 bundleSettings->bundleTargetBody());
 
+      bool isIsisObservation = true;
+
+      // This NULL check is needed solely for the unit test
```
Comment (jessemapel) [inline isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp:131]: How much work would it be to update the unit test and remove these if camera!=NULL checks?

Comment (krlberry) [reply isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp:131]: I'm not entirely sure. I can give it a try and timebox it?

Comment (jessemapel) [reply isis/src/control/objs/BundleUtilities/BundleObservationVector.cpp:131]: This looks like it's working so how about we merge this and you can put up a separate PR that tries to remove these?

**Review by jessemapel — APPROVED** [review]

**Pull request merged**